src/Umbraco.Web.Common/Extensions/UmbracoCoreServiceCollectionExtensions.cs

```diff
@@ -122,8 +174,17 @@ public static IServiceCollection AddUmbracoCore(this IServiceCollection services
             if (container is null) throw new ArgumentNullException(nameof(container));
             if (entryAssembly is null) throw new ArgumentNullException(nameof(entryAssembly));
 
+            services.AddSingleton<IDbProviderFactoryCreator>(x => new DbProviderFactoryCreator(
```

Comment:
This is another case where we are resolving things from a pre-resolved container. This means that this instance of IDbProviderFactoryCreator and all of it's dependencies at this stage will not be the same instances as ones resolved after the real container is build so they aren't actually "singletons." There is an ugly work around for this that I've created which is to re-register all of these types in the container as a delegate that points to the pre-resolved services but of course that is a nasty hack. 

I'm unsure how we can work around this but i'll see what i can find.

Reply:
It seems that we are doing this purely to use this object for the SqlMainDomLock BUT we don't want to do that, that is why we manually created these things before, just like we do in v8. We don't support SqlMainDomLock with SqlCe, only SQL server which was configured before. I don't think we need this here.